src/data/int/cast.lean

```diff
@@ -6,12 +6,27 @@ Authors: Mario Carneiro
 import data.int.basic
 import data.nat.cast
 
+/-!
+# Cast of integers
+
+This file defines the *canonical* homomorphism from the integers into a type `α` with `0`,
+`1`, `+` and `-` (typically an `add_group`).
```

Comment:
```suggestion
`1`, `+` and `-` (typically an `ring`).
```

Since an add_group might not have a `1`, and the existing docstrings already reference rings and ring-like structures